p2p/udp.go

```diff
@@ -191,33 +225,57 @@ func (upm *udpProtocolMessage) Data() service.Data {
 }
 
 // processUDPMessage processes a udp message received and passes it to the protocol, it adds related p2p metadata.
-func (mux *UDPMux) processUDPMessage(sender p2pcrypto.PublicKey, fromaddr net.Addr, buf []byte) error {
-	mux.logger.Debug("Processing message from %v, %v, len:%v", sender.String(), fromaddr.String(), len(buf))
-	msg := &ProtocolMessage{}
-	err := types.BytesToInterface(buf, msg)
+func (mux *UDPMux) processUDPMessage(msg inet.IncomingMessageEvent) error {
+	if msg.Message == nil || msg.Conn == nil {
+		return ErrBadFormat1
+	}
+
+	// protocol messages are encrypted in payload
+	// Locate the session
+	session := msg.Conn.Session()
+
+	if session == nil {
+		return ErrNoSession
+	}
+
+	rawmsg, _, err := p2pcrypto.ExtractPubkey(msg.Message)
+
+	if err != nil {
+		return err
+	}
+
+	decPayload, err := session.OpenMessage(rawmsg)
+	if err != nil {
+		mux.logger.Warning("failed decrypting message err=%v", err)
+		return ErrFailDecrypt
+	}
+
+	pm := &ProtocolMessage{}
+	err = types.BytesToInterface(decPayload, pm)
 	if err != nil {
-		return errors.New("could'nt deserialize message")
+		mux.logger.Error("proto marshaling err=", err)
```

Comment:
```suggestion
		mux.logger.Error("deserialization err=", err)
```
(this is XDR deserialization from bytes into a struct defined using protobuf, or am I missing something?)

Reply:
it was just leftovers from when we used protobufs